parlai/chat_service/services/websocket/websocket_manager.py

```diff
@@ -107,6 +108,16 @@ def _done_callback(fut):
                 agent_state = self.get_agent_state(agent.id)
                 agent_state.set_active_agent(agent_state.get_overworld_agent())
 
+        # Monitor for crashed overworld futures
+        for agent_id, overworld_fut in self.agent_id_to_overworld_future.items():
+            if overworld_fut.done():
+                try:
```

Comment:
nit: this logic is a bit vague here. can we just check that the future didn't finish with an exception by checking `overworld_fut.exception()` directly?

Reply:
I added a done callback to the overworld futures that checks for exceptions and handles completed overworlds, so this block of code is no longer needed